src/au-table.js

```diff
@@ -101,14 +101,17 @@ export class AureliaTableCustomAttribute {
 
     doFilter(toFilter) {
         let filteredData = [];
-
+        
         for (let next of toFilter) {
             for (let key of this.filterKeys) {
-                let value = next[key].toString();
 
-                if (value.toLowerCase().indexOf(this.filterText.toLowerCase()) > -1) {
-                    filteredData.push(next);
-                    break;
+                if( next[key]){
```

Comment:
great point, although `next[key] != null` also checks for undefined due to coercion.  I'll fix this in the pull request.